### DISCUSSION TITLE
How does the ivfflat index use search_preassigned

### DISCUSSION THREAD

**Discussion by lambda7xx** [facebookresearch/faiss#3601] [General]
# Summary

# Platform

OS: 

Faiss version: 

Installed from:  

Faiss compilation options: 

Running on:
- [x] CPU
- [ ] GPU

Interface: 
- [ ] C++
- [x] Python

# Reproduction instructions

**Comment by lambda7xx** [comment]
my faiss-cpu version is 1.8. my code is below
```
index.search_preassigned(emb.shape[0], emb, nnn, candidate_cluster, None, topk_distance, topk_indices)
```
I try to use the `search_preassigned` API for ivfflat. I want to search the top-nnn vector in the candidate cluster and set the `centroid_dis` as None(https://faiss.ai/cpp_api/struct/structfaiss_1_1IndexIVFFlat.html). Base on the document(https://faiss.ai/cpp_api/struct/structfaiss_1_1IndexIVFFlat.html), it has the `search_preassigned`.

but my error is

```
TypeError: handle_Index.<locals>.replacement_search_preassigned() takes 5 positional arguments but 8 were given
type(self.index):<class 'faiss.swigfaiss_avx2.IndexIVFFlat'>
```

**Comment by ndtreviv** [comment]
I recently upgraded FAISS and had the same/similar error:
```
in ivf_search_preassigned
index_ivf.search_preassigned(
handle_Index.<locals>.replacement_search_preassigned() takes 5 positional arguments but 9 were given
```

**Reply by lambda7xx** [reply]
> I recently upgraded FAISS and had the same/similar error:
> 
> ```
> in ivf_search_preassigned
> index_ivf.search_preassigned(
> handle_Index.<locals>.replacement_search_preassigned() takes 5 positional arguments but 9 were given
> ```

same error. They do not support for flat index. You can try Ivf-pq index
